src/include/miopen/solver.hpp

```diff
@@ -146,6 +148,33 @@ struct SolverBase
     ///                          const PerformanceConfig& config) const;
 };
 
+template<class Context>
+struct SolverMixin : SolverBase
```

Comment:
The mixin should take the Derived class as a template parameter, so it can implement `SolverDbId` and possibly auto registration in the future:

```cpp
template<class Derived, class Context>
struct SolverMixin : SolverBase
```

Reply:
This is a good suggestion in my opinion. Let's add it after accepting this PR